File: src/ios/darkmode.m
#import "darkmode.h"
#import <Foundation/Foundation.h>

@interface darkmode () 

@property (nonatomic) CDVInvokedUrlCommand* command;

@end

@implementation darkmode
- (void)darkmode:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
 
	 if (@available(iOS 13.0, *)) {

		if( self.traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark ){
		   //is dark
		}else{
			//is light

		}
	}
    NSString* compresssedstring = [[NSAppearance currentAppearance] name];
	bool isDark = [appearanceName containsString:@"Dark"];

    if (isDark) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Dark"];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
    }

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
	
}

@end

File: src/ios/darkmode.swift
@available(iOS 13, )
@objc(darkmode) class darkmode  CDVPlugin {
  @objc(darkmode)
  func darkmode(command CDVInvokedUrlCommand) {
    let isDark = UIScreen.main.traitCollection.userInterfaceStyle == .dark
    
    let pluginResult = CDVPluginResult(status CDVCommandStatus_OK, messageAsString true)
    
    self.commandDelegate!.send(pluginResult, callbackId command.callbackId)
  }
}